src/main/frontend/components/query/builder.cljs

```diff
@@ -0,0 +1,451 @@
+                  (reset! *property (keyword value)))))
+
+            "property-value"
+            (let [values (cons "Select all" (db-model/get-property-values @*property))]
+              (select values
+                (fn [value]
+                  (let [x (if (= value "Select all")
+                            [:property @*property]
+                            [:property @*property value])]
+                    (reset! *property nil)
+                    (append-tree! *tree opts loc x)))))
+
+            "sample"
+            (select (range 1 101)
+              (fn [value]
+                (append-tree! *tree opts loc [:sample (util/safe-parse-int value)])))
+
+            "task"
+            (select db-default/built-in-markers
+              (fn [value]
+                (append-tree! *tree opts loc [:task value])))
+
+            "priority"
+            (select db-default/built-in-priorities
+              (fn [value]
+                (append-tree! *tree opts loc [:priority value])))
+
+            "page"
+            (let [pages (sort (db-model/get-all-page-original-names repo))]
+              (select pages
+                (fn [value]
+                  (append-tree! *tree opts loc [:page value]))))
+
+            "page reference"
+            (let [pages (sort (db-model/get-all-page-original-names repo))]
+              (select pages
+                (fn [value]
+                  (append-tree! *tree opts loc [:page-ref value]))
+                {}))
+
+            "full text search"
+            (search (fn [v] (append-tree! *tree opts loc v))
+                    (:toggle-fn opts))
+
+            "between"
+            (between (merge opts
+                            {:tree *tree
+                             :loc loc
+                             :clause clause}))
+
+            nil)])
+       [:div
+        (when-not @*find
+          [:div.flex.flex-row.items-center.p-2
+           [:div.ml-2 "Find: "]
+           (page-block-selector *find)])
+        (when-not @*find
+          [:hr.m-0])
+        (select
+          (map name filters-and-ops)
+          (fn [value]
+            (cond
+              (= value "all page tags")
+              (append-tree! *tree opts loc [:all-page-tags])
+
+              (operator? value)
+              (append-tree! *tree opts loc [(keyword value)])
+
+              :else
+              (reset! *mode value)))
+          {:input-default-placeholder "Add filter/operator"})])]))
+
+(rum/defc add-filter
+  [*find *tree loc clause]
+  (ui/dropdown
+   (fn [{:keys [toggle-fn]}]
+     [:a.flex.add-filter {:title "Add clause"
+                          :on-click toggle-fn}
+      (ui/icon "plus" {:style {:font-size 20}})])
+   (fn [{:keys [toggle-fn]}]
+     (picker *find *tree loc clause {:toggle-fn toggle-fn}))
+   {:modal-class (util/hiccup->class
+                  "origin-top-right.absolute.left-0.mt-2.ml-2.rounded-md.shadow-lg")}))
+
+(declare clauses-group)
+
+(defn- dsl-human-output
```

Comment:
Seeing some bugs with this. Happy to help fix em on Monday if u'd like:
- task and priority only show the first argument to them e.g. (task TODO DOING) only shows TODO
- namespace op displays as `[:page-ref "abc"]`
- property op when selecting all looks weird

Reply:
> namespace op displays as [:page-ref "abc"]

Failed to reproduce this too.

> property op when selecting all looks weird

Do you mean the `key: *` part?

> task and priority only show the first argument to them e.g. (task TODO DOING) only shows TODO

[bbf555c](https://github.com/logseq/logseq/pull/8774/commits/bbf555c7f628784dcbba212d0f6e156cd2387008) addressed this problem.
Users can select multiple choices now. 
https://www.loom.com/share/a69babc6ae7a445eabdbf2dcaab0c68f

Reply:
> Failed to reproduce this too.

Also failed to reproduce 😛 

> Users can select multiple choices now.

Thanks! Works and looks great

> Do you mean the key: * part?

Maybe but I'm unable to repro it as the "Select all" choice no longer shows up for property filters

Reply:
> Maybe but I'm unable to repro it as the "Select all" choice no longer shows up for property filters

Nm. I'm able to repro. `key: *` is what I see. I saw something weirder before but can't repro it now. We're good here :)

Reply:
Cool, I replaced `*` with `ALL` hopefully to make it a bit user-friendly. Also, I added `Live query for pages` to avoid confusion with queries for blocks.
<img width="820" alt="CleanShot 2023-03-13 at 23 11 22@2x" src="https://user-images.githubusercontent.com/479169/224744391-a9421616-e425-40a2-a88a-bc68baa71204.png">